content/docs/migration_1.0/Pico.md

```diff
@@ -0,0 +1,263 @@
+
+- Zenoh 0.11.x
+
+```c
+const char *keyexpr = "example/demo/*";
+z_owned_subscriber_t sub = z_declare_subscriber(z_loan(session), z_keyexpr(keyexpr), z_move(callback), NULL);
+if (!z_check(sub)) {
+    printf("Unable to declare subscriber.\n");
+    return -1;
+}
+```
+
+- Zenoh 1.0.0
+
+```c
+const char *keyexpr = "example/demo/*";
+z_owned_subscriber_t sub;
+z_view_keyexpr_t ke;
+z_view_keyexpr_from_str(&ke, keyexpr);
+if (z_declare_subscriber(&sub, z_loan(session), z_loan(ke), z_move(callback), NULL) < 0) {
+    return -1;
+}
+```
+
+## Payloads and serialization
+
+We changed how payloads are handled. Before you would pass the pointer and the size of your data and now everything must be serialized into `z_owned_bytes_t`. 
+
+We have added a number of serialization functions from primitive types to and from `z_owned_bytes_t` to simplify this.
+
+- Zenoh 0.11.x
+
+```c
+char *value = "Some data to publish on Zenoh";
+
+if (z_put(z_loan(session), z_keyexpr(ke), (const uint8_t *)value, strlen(value), NULL) < 0) {
+    return -1;
+}
+```
+
+- Zenoh 1.0.0
+
+```c
+char *value = "Some data to publish on Zenoh";
+z_owned_bytes_t payload;
+z_bytes_serialize_from_str(&payload, value);
+
+if (z_put(z_loan(session), z_loan(ke), z_move(payload), NULL) < 0) {
+  return -1;
+}
+```
+
+## Attachment
+
+Attachments are now a second payload and must also be serialized to `z_owned_bytes_t`. 
+
+We have a serialization function to encode the key/pair value but nothing prevents use to serialize an another type as attachment.
+
+- Zenoh 0.11.x
+
+```c
+char *value = "Some data to publish on Zenoh";
+
+z_put_options_t options = z_put_options_default();
+z_owned_bytes_map_t map = z_bytes_map_new();
+z_bytes_map_insert_by_alias(&map, _z_bytes_wrap((uint8_t *)"test", 2), _z_bytes_wrap((uint8_t *)"attachement", 5));
+options.attachment = z_bytes_map_as_attachment(&map);
+
+if (z_put(z_loan(session), z_keyexpr(ke), (const uint8_t *)value, strlen(value), NULL) < 0) {
+    return -1;
+}
+
+if (z_put(z_loan(s), z_keyexpr(keyexpr), (const uint8_t *)value, strlen(value), &options) < 0) {
+    return -1;
+}
+
+z_bytes_map_drop(&map);
+```
+
+- Zenoh 1.0.0
+
+```c
+
+char *value = "Some data to publish on Zenoh";
+z_owned_bytes_t payload;
+z_bytes_serialize_from_str(&payload, value);
+
+// It's up to the user to create the item serializer function
+z_publisher_put_options_t options;
+z_publisher_put_options_default(&options);
+z_owned_bytes_t attachment;
+z_bytes_serialize_from_iter(&attachment, create_attachment_iter, (void *)&ctx);
+options.attachment = z_move(attachment);
+
+if (z_put(z_loan(session), z_loan(ke), z_move(payload), &options) < 0) {
+  return -1;
+}
+```
+
+## Encoding
+
+How encoding is handled has been reworked:  While before you would use an enum id and a value, now you simply need to register your encoding metadata from a string with `z_encoding_from_str`. 
```

Comment:
A list of constant should also be defined and provided. To double check this is the case for both `zenoh-c` and `zenoh-pico`.

Reply:
Pico provides a list of encoding constants in an enum. 
@Mallets 
Do you want that i add that list of 21 variants to the guide itself ?

Reply:
Added a link in Pico to the Enum variants